interfaces/src/dotty/tools/dotc/interfaces/SimpleReporter.scala

```diff
@@ -0,0 +1,19 @@
+package dotty.tools.dotc.interfaces
+
+import scala.annotation.binaryCompatible
+
+
+/** Report errors, warnings and info messages during the compilation process
+ *
+ * You should implement this interface if you want to handle the diagnostics
+ * returned by the compiler yourself.
+ *
+ * See the method `process` of `dotty.tools.dotc.Driver` for more information.
+ */
+@binaryCompatible
+trait SimpleReporter {
```

Comment:
I retract my comment, the code I was compiling had a non-abstract method, that's why these methods existed. (`$init$` seems to always be generated in dotty but that should be considered a bug)